beetmoverscript/script.py

```diff
@@ -292,6 +292,14 @@ def generate_balrog_info(context, artifact_pretty_name, locale, destinations, fr
     }
     if from_buildid:
         data["from_buildid"] = from_buildid
+        if is_promotion_action(context.action):
+            partials = get_partials_props(context.task)
+            # TODO: improve search by a dedicated dict with buildid as key
+            for p in partials.values():
+                if p['buildid'] == str(from_buildid):
+                    data['previousVersion'] = p['previousVersion']
+                    data['previousBuildNumber'] = p['previousBuildNumber']
```

Comment:
Do we want a `break` here?

Also, do we have anything validating the `get_partials_props` schema? That may belong in `get_partials_props`, or in the release task schema as an optional field. This doesn't block, but may help in the future.